src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -256,10 +262,25 @@ private static String getSkippedMessage(Element testCase) {
         return message;
     }
 
-    public String getDisplayName() {
+    public String getTransformedTestName() {
         return TestNameTransformer.getTransformedName(testName);
     }
 
+    public String getDisplayName() {
+        // Only prepend the enclosing flow node names if there are any and the run this is in has multiple blocks directly containing
+        // test results.
+        if (!getEnclosingFlowNodeNames().isEmpty()) {
```

Comment:
>  here means parents

Yes, more specifically *parent.*   Multiple parents are only possible for the BlockEndNode ending  a set of Parallel branches -- and because that's a block end node, you look at thee BlockStartNode and its parent.